agent-control/src/agent_type/render/renderer.rs

```diff
@@ -181,6 +181,7 @@ pub(crate) mod tests {
     use assert_matches::assert_matches;
     use fs::directory_manager::DirectoryManagementError;
     use mockall::{mock, predicate};
+    use std::collections::VecDeque;
```

Comment:
If I'm understanding the test well this shouldn't be needed no? I don't have a strong opinion since we are in the test module, but it seems it can be done with `Vec`, with mutable slice indexes or leave mutability to iterators combining the two collections (e.g. [`zip`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.zip)).

Reply:
Well I simpliefied it with an iterator and next() in last commit.